Content.Server/Resist/ResistLockerSystem.cs

```diff
@@ -0,0 +1,115 @@
+using Robust.Shared.Player;
+using Robust.Shared.Containers;
+using Content.Server.Popups;
+using Robust.Shared.Localization;
+
+namespace Content.Server.Resist;
+
+public class ResistLockerSystem : EntitySystem
+{
+    [Dependency] private readonly DoAfterSystem _doAfterSystem = default!;
+    [Dependency] private readonly PopupSystem _popupSystem = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<ResistLockerComponent, RelayMovementEntityEvent>(OnRelayMovement);
+        SubscribeLocalEvent<ResistLockerComponent, ResistDoAfterComplete>(OnDoAfterComplete);
+        SubscribeLocalEvent<ResistLockerComponent, ResistDoAfterCancelled>(OnDoAfterCancelled);
+        SubscribeLocalEvent<ResistLockerComponent, EntRemovedFromContainerMessage>(OnRemovedFromContainer);
+    }
+
+    private void OnRelayMovement(EntityUid uid, ResistLockerComponent component, RelayMovementEntityEvent args)
+    {
+        TryComp(uid, out EntityStorageComponent? storageComponent);
+        if (!Resolve(uid, ref storageComponent))
+            return;
+
+        if (!component.IsResisting)
+        {
+            if (TryComp<LockComponent>(uid, out var lockComponent) && lockComponent.Locked || storageComponent.IsWeldedShut)
+            {
+                AttemptResist(args.Entity, uid, storageComponent, component);
+            }
+        }
+    }
+
+    private void AttemptResist(EntityUid user, EntityUid target, EntityStorageComponent? storageComponent, ResistLockerComponent? resistLockerComponent)
+    {
+        if (!Resolve(target, ref storageComponent, ref resistLockerComponent))
+            return;
+
+        resistLockerComponent.CancelToken = new();
+        var doAfterEventArgs = new DoAfterEventArgs(user, resistLockerComponent.ResistTime, resistLockerComponent.CancelToken.Token, target)
+        {
+            BreakOnTargetMove = false,
+            BreakOnUserMove = true,
+            BreakOnDamage = true,
+            BreakOnStun = true,
+            NeedHand = false, //No hands 'cause we be kickin'
+            TargetFinishedEvent = new ResistDoAfterComplete(user, target),
+            TargetCancelledEvent = new ResistDoAfterCancelled(user)
+        };
+
+        resistLockerComponent.IsResisting = true;
+        _popupSystem.PopupEntity(Loc.GetString("resist-locker-component-start-resisting"), user, Filter.Entities(user));
+        _doAfterSystem.DoAfter(doAfterEventArgs);
+    }
+
+    private void OnDoAfterComplete(EntityUid uid, ResistLockerComponent component, ResistDoAfterComplete ev)
+    {
+        component.IsResisting = false;
+
+        if (TryComp<EntityStorageComponent>(uid, out var storageComponent))
+        {
+            if (storageComponent.IsWeldedShut)
+                storageComponent.IsWeldedShut = false;
+
+            if (TryComp<LockComponent>(ev.Target, out var lockComponent))
+            {
+                lockComponent.Locked = false;
+            }
```

Comment:
You shouldn't just directly set the lock component's state. The lock component should really have a `[Friend(typeof(LockSystem))]` to prevent this. As to why: you can see in the `LockSystem`'s that it calls
```cs
if (EntityManager.TryGetComponent(lockComp.Owner, out AppearanceComponent? appearanceComp))
{
    appearanceComp.SetData(StorageVisuals.Locked, true);
}

RaiseLocalEvent(lockComp.Owner, new LockToggledEvent(true));
```
whenever the lock is toggled, so that the visuals get updated & other systems get notified. Unfortunately `LockSystem` doesn't currently have a `SetLocked` function, just functions for users attempting to (un)lock things. So you probably need to create that function and call that instead of directly setting `lockComponent.Locked = false;`

Setting `IsWeldedShut` directly works fine because storageComponent is old code that still has logic in the component's setter

Reply:
Ah ShadowCommander already just brought this up

Reply:
Yea, i've moved the actual unlocking logic to an Unlock function, and TryUnlock just checks the user for appropriate access before calling Unlock. Thanks!